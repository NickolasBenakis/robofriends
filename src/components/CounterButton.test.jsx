import React from 'react';
import { shallow } from 'enzyme';
import CounterButton from './CounterButton';

it('renders without crashing', () => {
    const mockColor = 'red';
    expect(shallow(<CounterButton color={mockColor} />)).toMatchSnapshot();
});

it('correctly increments the counter', () => {
    const mockColor = 'red';
    const wrapper = shallow(<CounterButton color={mockColor} />)
    expect(wrapper).toMatchSnapshot();
    wrapper.find('[id="counter"]').simulate('click');
    expect((wrapper.state())).toEqual({ count: 1 })
    wrapper.find('[id="counter"]').simulate('click');
    wrapper.find('[id="counter"]').simulate('click');
    wrapper.find('[id="counter"]').simulate('keypress');
    expect((wrapper.state())).toEqual({ count: 3 })
    expect((wrapper.props().color)).toEqual("red");

});

it('checks if shouldCOmponment update works', () => {
    const mockColor = 'red'
    const wrapper = shallow(<CounterButton color={mockColor} />)

    const noUpdate = wrapper.instance().shouldComponentUpdate({}, { count: 0 })
    expect(noUpdate).toEqual(false)

    const shouldUpdate = wrapper.instance().shouldComponentUpdate({}, { count: 1 })
    expect(shouldUpdate).toEqual(true)
})