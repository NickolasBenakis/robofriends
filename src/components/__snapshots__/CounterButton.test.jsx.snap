// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correctly increments the counter 1`] = `ShallowWrapper {}`;

exports[`renders without crashing 1`] = `ShallowWrapper {}`;
